crates/compiler/build/src/link.rs

```diff
@@ -290,15 +267,8 @@ pub fn build_zig_host_wasm32(
             "build-obj",
             zig_host_src,
             emit_bin,
-            "--mod",
-            &format!("glue::{}", find_zig_glue_path().to_str().unwrap()),
-            "--deps",
-            "glue",
             // include the zig runtime
             // "-fcompiler-rt",
-            // include libc
-            "--library",
-            "c",
```

Comment:
This was causing issues with `zig build-obj` when exporting `main`. We had an error saying duplicate symbol. 

I believe the cause of this is that now zig 0.13.0 bundles wasi-libc, and so if we include `-lc` we now have two main functions, one from wasi-libc for the WASI implementation because we build the host at `-target wasm32-wasi` and also from out `host.zig`.

Reply:
@bhansconnect this may be relevant context for our discussion re export for `wasm_linking_test_host.zig`